### ISSUE TITLE
add vercel

### ISSUE THREAD

**Issue by Trim21** [quisquous/cactbot#3053]
https://github.com/quisquous/cactbot/pull/3049

https://vercel.com/ is a tool that could build each pr and serving deploy static files public.

For example, after adding this tool, each pr will get a unique URL like "HTTPS://cactbot-pr-3049.someplace.domain/" with builded static files served (`dist` dir for us).

So for previewing a pr, we can try builded files like they are hosted on gh-pages without checkout (may be npm I) , and build.

I keep my cactbot version in act seprated with development environment, and just try distribution in browsers(but they share same act backend), looks like we are having different workflow.

This is a example https://github.com/DIYgod/RSSHub/pull/7467#issuecomment-830872979

(This deployed URL of this pr may be outdated because it's a old PR)

**Comment by panicstevenson** [comment]
I have somewhat mixed feelings about this since this is a separate thing trying to sell us a product, but the pricing tier seems reasonable (free for 100 deployments/day, 10 deployment hours/month); it definitely feels unlikely that we would exceed the free tier's numbers. One thing I didn't see is how long the PR build lasts; I think that's important to my overall assessment. Overall, I'm "okay" with this idea since it feels like the pricing tiers would never hinder us and it seems neat.

However, I'm curious how much of an issue this is, and if we couldn't solve it ourselves. We build the plugin on every plugin PR thanks to MaikoTan, and you can already build to your own gh-pages so long as you push to `main` on your fork (see: https://github.com/quisquous/cactbot/pull/1883#issuecomment-710700736 as an example). What would be the use here other than a smoke test to confirm things still work? I would argue those should become smoke tests (which we probably are desperate for).

Thinking about other things that might be neat/interesting/helpful would be a way to have raidemulator be built against a branch that a non-contributor could upload a log against (without having to clone the repository or anything) and explicitly see if their trigger issue is remedied. I don't think we're at that point yet, but this may be useful/a solution to that if we decide that's something we want in the future. I'm curious to hear what other people think. I think I'm overall warm to the idea, but I don't know if we have enough builds that come across that I would go change my ACT URL and go test for each one, if that makes sense?

**Comment by quisquous** [comment]
> Thinking about other things that might be neat/interesting/helpful would be a way to have raidemulator be built against a branch that a non-contributor could upload a log against

I do think once #2942 lands we can have people use the gh-pages branch for testing against raidemulator.

> I don't know if we have enough builds that come across that I would go change my ACT URL and go test for each one, if that makes sense?

Yeah, I think this is the issue for me, that changing the ACT URL for overlays is cumbersome in my workflow.  I usually have the WebPack dev server running, so if I just pull down a PR and switch to that branch, then everything is working automatically.

I think this just feels like a lot of investment that only saves checking out a pr locally.  Maybe I just don't understand your workflow well enough to understand what this is saving.

**Comment by valarnin** [comment]
Another concern is, since the domain name changes every time there's a new PR or whatever, the indexeddb data for raidemulator would sit on disk for those domains until the browser auto-cleared it or something, it wouldn't transfer from PR to PR.
